wakebox/main.cpp

```diff
@@ -73,12 +83,41 @@ int main(int argc, char *argv[])
 	}
 	ifs.close();
 
+	// Parse the params file as JSON.
+	JAST jast;
+	if (!JAST::parse(json, std::cerr, jast))
+		return 1;
+
+	// Extract the 'command' from the input json.
+	std::vector<char *> command;
+	for (auto &x : jast.get("command").children)
+		command.push_back(const_cast<char*>(x.second.value.c_str()));
```

Comment:
The above leaks the bug from execv all the way to the user.
Let's use a std::vector<std::string> instead.

Then, at execve call site, we convert it to std::vector<const char *> (which does not need a const_cast to convert).
At call, do: execve(... const_cast<char * const *>(vec.data()), ...)
And a comment indicating the const_cast is to work around the buggy POSIX execve API definition.